### PR TITLE
Makes time agents a latejoin ruleset too

### PR DESCRIPTION
**Pull request by ghost**
[gamemode]
:cl:
 * rscadd: People can now sometimes become time agents instead of joining the round, if rulesets allow it. (with an option not to, like ninjas)

### CODE REVIEW

**Comment by Eneocho** [comment]
Time agent is built around having only one guy (and the eventual evil twin), can this fire more than once?

**Comment by SECBATON-GRIFFON** [comment]
> Time agent is built around having only one guy (and the eventual evil twin), can this fire more than once?

No

**Comment by ghost** [comment]
> Time agent is built around having only one guy (and the eventual evil twin), can this fire more than once?

No

**Comment by Eneocho** [comment]
Can the evil twin fire from this? As in, a latejoin becomes the evil twin if the required 5 minutes have passed? Or does it still call for observers to become?

File: code/datums/gamemode/dynamic/dynamic_rulesets_latejoin.dm [hunk 1]
```diff
@@ -239,3 +239,49 @@
...
+//                                          //
+//               TIME AGENT                 //
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/latejoin/time_agent
+	name = "Time Agent Anomaly"
+	role_category = /datum/role/time_agent
+	required_candidates = 1
+	weight = 4
```
Comment (gurfan) [inline code/datums/gamemode/dynamic/dynamic_rulesets_latejoin.dm]: magic number

**Comment by ghost** [comment]
> Can the evil twin fire from this? As in, a latejoin becomes the evil twin if the required 5 minutes have passed? Or does it still call for observers to become?

Still calls for observers, that's why even the latejoin check makes sure there is some.

File: code/datums/gamemode/dynamic/dynamic_rulesets_latejoin.dm (hunk 1 above)
Comment (ghost) [reply code/datums/gamemode/dynamic/dynamic_rulesets_latejoin.dm]: Fixed

**Pull request merged**